src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -2283,4 +2287,20 @@ private function getClasses(Definition $definition, string $id): array
 
         return $classes;
     }
+
+    private function isProxyCandidate(Definition $definition, bool &$asGhostObject = null): ?Definition
+    {
+        $asGhostObject = false;
+
+        if (!$definition->isLazy() || ($proxyDumper = $this->getProxyDumper()) instanceof NullDumper) {
+            return null;
+        }
+
+        $bag = $this->container->getParameterBag();
+        $definition = (clone $definition)
+            ->setClass($bag->resolveValue($definition->getClass()))
+            ->setTags($bag->resolveValue($definition->getTags()));
```

Comment:
Resolving tags might be an issue, because they are generally not resolved, which means that this could trigger failures that would be unexpected.